zerver/actions/streams.py

```diff
@@ -1357,3 +1371,40 @@ def do_change_stream_group_based_setting(
     transaction.on_commit(
         lambda: send_event(stream.realm, event, can_access_stream_user_ids(stream))
     )
+
+
+@transaction.atomic(durable=True)
+def do_change_push_notifications_enabled(
+    stream: Stream, acting_user: UserProfile, push_notifications_enabled: bool
```

Comment:
Would be better to keep `acting_user` as mandatory keyword argument like we do for other similar functions.